src/braket/circuits/circuit.py

```diff
@@ -412,10 +434,30 @@ def add_instruction(
             # non single qubit operator with target, add instruction with target
             instructions_to_add = [instruction.copy(target=target)]
 
+        if self._check_for_params(instruction):
+            self._parameterized = True
+            self._parameters.add(instruction.operator.parameter)
         self._moments.add(instructions_to_add)
 
         return self
 
+    def _check_for_params(self, instruction: Instruction):
+        """
+        This checks for free parameters in an :class:{Instruction}.
+
+        Args:
+            instruction: The instruction to check for a :class:{FreeParameter}.
+
+        Returns: True if the angle is not a number. False otherwise.
+        """
+        # Check for Angled Gates.
+        if (
+            issubclass(type(instruction.operator), QuantumOperator)
+            and instruction.operator.parameterized
+        ):
+            return True
+        return False
```

Comment:
Another reason why `parameter` should live with AngledGate rather than QuantumOperator.